tarofreighter/wallet.go

```diff
@@ -280,3 +448,148 @@ func inputAnchorPkScript(assetInput *AnchoredCommitment) ([]byte, []byte,
 	pkScript, err := taroscript.PayToTaprootScript(anchorPubKey)
 	return pkScript, taroScriptRoot[:], err
 }
+
+// adjustFundedPsbt takes a funded PSBT which may have used BIP 69 sorting, and
+// creates a new one with outputs shuffled such that the change output is the
+// last output.
+func adjustFundedPsbt(pkt *tarogarden.FundedPsbt, anchorInputValue int64) {
+	// If there is no change there's nothing we need to do.
+	changeIndex := pkt.ChangeOutputIndex
+	if changeIndex == -1 {
+		return
+	}
+
+	// Store the script and value of the change output.
+	maxOutputIndex := len(pkt.Pkt.UnsignedTx.TxOut) - 1
+	changeOutput := pkt.Pkt.UnsignedTx.TxOut[changeIndex]
+
+	// Overwrite the existing change output, and restore in at the
+	// highest-index output.
+	pkt.Pkt.UnsignedTx.TxOut[changeIndex] = createDummyOutput()
+	pkt.Pkt.UnsignedTx.TxOut[maxOutputIndex].PkScript = changeOutput.PkScript
+	pkt.Pkt.UnsignedTx.TxOut[maxOutputIndex].Value = changeOutput.Value
+
+	// Since we're adding the input of the anchor output of our prior asset
+	// later, we need to add this value here, so we don't lose the amount
+	// to fees.
+	pkt.Pkt.UnsignedTx.TxOut[maxOutputIndex].Value += anchorInputValue
```

Comment:
I think we'll also want to revisit some of this logic here w/ the amounts and fees as well. When I send on the CLI, the posted chain fees look to be 10x or so higher than I'd expect.

Reply:
Yes, definitely! Should be a bit easier now with the new code in place.